Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp

```diff
@@ -107,42 +107,37 @@ void Init()
 	_dbg_assert_msg_(WII_IPC_HLE, g_DeviceMap.empty(), "DeviceMap isn't empty on init");
 	CWII_IPC_HLE_Device_es::m_ContentFile = "";
 
-	for (IWII_IPC_HLE_Device*& dev : g_FdMap)
-	{
-		dev = nullptr;
-	}
-
 	u32 i = 0;
 	// Build hardware devices
-	g_DeviceMap[i] = new CWII_IPC_HLE_Device_usb_oh1_57e_305(i, "/dev/usb/oh1/57e/305"); i++;
-	g_DeviceMap[i] = new CWII_IPC_HLE_Device_stm_immediate(i, "/dev/stm/immediate"); i++;
-	g_DeviceMap[i] = new CWII_IPC_HLE_Device_stm_eventhook(i, "/dev/stm/eventhook"); i++;
-	g_DeviceMap[i] = new CWII_IPC_HLE_Device_fs(i, "/dev/fs"); i++;
+	g_DeviceMap[i++] = std::make_shared<CWII_IPC_HLE_Device_usb_oh1_57e_305>(i, "/dev/usb/oh1/57e/305");
```

Comment:
Is this a good idea? I don't think it does what you/we want; and might even be undefined behavior? Its probably a better idea to do ```i++``` elsewhere.